src/org/openstreetmap/josm/tools/Logging.java

```diff
@@ -50,6 +50,66 @@
     private static final Logger LOGGER = Logger.getAnonymousLogger();
     private static final RememberWarningHandler WARNINGS = new RememberWarningHandler();
 
+    /**
+     * A {@link ConsoleHandler} with a couple of extra features, allowing it to be targeted at an
+     * an arbitrary {@link OutputStream} which it can be asked to reaquire the reference for on demand
+     * through {@link #reacquireOutputStream()}. It can also prevent a LogRecord's output if a
+     * specified {@code prioritizedHandler} would have outputted it.
+     */
+    public static class ReacquiringConsoleHandler extends ConsoleHandler {
```

Comment:
It does not seem restricted to logging. This class could be standalone.

Reply:
Hmm it inherits from `logging.ConsoleHandler` - I guess I could try and base it on something far more generic, but that thing would probably be so generic as to not be hugely useful and I think the resulting class would be significantly more complicated...

Reply:
OK we can keep it here then 👍